### ISSUE TITLE
Possible to make Keycode optional?

### ISSUE THREAD

**Issue by VHouse1926** [bitfocus/companion-module-lgtv-display#12]
Greetings! Thank you for writing this module. I'm wondering if it's possible to make the keycode optional or change the accepted format? I also noticed the module might need to be updated for Companion 3.x

I have an LG 55LH575A. I was able to get to the hidden menu and turn on network control. However, i don't have the option in those settings to generate a keycode like some LG TVs. I can put a 3 or four-digit access code in manually, but I'm not positive if it's used in the handshake at all.

Attached is the manual which unfortunately does not list anything about IP control. only shows the RS232 commands. 
Thank you for your time!

[MFL69272417_02+MFL69289418_01+RS.pdf](https://github.com/user-attachments/files/16068857/MFL69272417_02%2BMFL69289418_01%2BRS.pdf)

**Comment by Jibun-no-Kage** [comment]
Unfortunately, at least with the LG TVs I have access to, all require the key to communicate, i.e. receive controls, this seems to be a change that happened recently about 2022 and later, best I can determine.  ViZIO and ROKU on some models still have an open REST-API, but this may not continue as industrial units and consumer units become the same design, where industrial units must have some method to stop 'bad actor' take over.  For example some industrial units have option to completely disable infra-red receiver so someone, say in an airport, cannot just take control of the TV, change the keycode, etc.  Some industrial units even have custom connectors for cabling, so HDMI uses a special non-standard connector... again to remove the ability of direct connect to the TV and using say CEC protocol to take control of the device.

**Comment by bryce-seifert** [comment]
Yes, the keycode is a requirement for connecting to all modern LG TVs. so it can't be removed.

Also, this module only worked with LG TVs 2018+ as those have the IP control capabilities, so I think regardless it would not work with the specified model as that TV seems to pre-date that cutoff.
